core/src/main/scala/zio/config/ConfigSource.scala

```diff
@@ -4,15 +4,21 @@ import zio.{ IO, ZIO }
 import zio.system.System.Live.system
 import java.{ util => ju }
 
-final case class ConfigValue[A](value: ::[A])
+final case class ConfigValue[A](value: ::[Option[::[A]]])
```

Comment:
Lists can exist 

[
   {
   key1: [ value1, value2],
   key2: [value2, value3]
  },
   {
   key1: [ value3, value4],
   key2: [value5, value6]
  },
]

This opens the space for dynamic map

example

```
KAFKA_BOOTSTRAP_SERVERS = "x"
KAFKA_CG_NAME = "y"
```

and we could do

```
nested("kafka")(map)
```

to obtain

```
Map(
   BOOTSTRAP_SERVERS -> x,
   CG_NAME -> y
 )
```

Reply:
I think in the long run, it may be simpler to migrate to `PropertyTree`. We can have helpers for "flat" property formats that make it easier to construct `ConfigSource`.